tlv/complex.go

```diff
@@ -0,0 +1,48 @@
+package tlv
+
+import (
+	"io"
+)
+
+// IndexStatus is an enum-like type that describes what state the indexed
+// outpoint is in. Currently only two possible values.
+type IndexStatus uint8
+
+const (
+	// OutpointOpen represents an outpoint that is open in the outpoint index.
+	OutpointOpen IndexStatus = 0
```

Comment:
do we want to include `Pending`/`PendingClosed` here?

Reply:
I thought no as then we have to add more places to keep track of the state. But if the change is minimal then I don't see why not.

Reply:
It's probably best to leave it off for now, as determining the other states requires fully decoding the channels, which means a lot more code needs to be isolated for this migration.